src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -82,6 +82,64 @@ public T get(final int... keys) {
         return (T)myData[keys[numNestedDimensions]];
     }
 
+    @SuppressWarnings("unchecked")
+    private T leaf(final int key, final Object[] array){
+        return array == null ? null : (T) array[key];
```

Comment:
This strikes me as a pretty weak argument -- you control the call site as well in this case, since `leaf()` et. al. are private. So you should handle the `ClassCastException` at the call site.